exporters/autoexport/exporter.go

```diff
@@ -0,0 +1,98 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package autoexport // import "go.opentelemetry.io/contrib/exporters/autoexport"
+
+import (
+	"context"
+	"os"
+
+	"go.opentelemetry.io/otel/sdk/trace"
+)
+
+const (
+	otelTracesExportersEnvKey = "OTEL_TRACES_EXPORTER"
+)
+
+type config struct {
+	fallbackExporter trace.SpanExporter
+}
+
+func newConfig(ctx context.Context, opts ...Option) (config, error) {
+	cfg := config{}
+	for _, opt := range opts {
+		cfg = opt.apply(cfg)
+	}
+
+	// if no fallback exporter is configured, use otlp exporter
+	if cfg.fallbackExporter == nil {
+		exp, err := SpanExporter(context.Background(), "otlp")
+		if err != nil {
+			return cfg, err
+		}
+		cfg.fallbackExporter = exp
+	}
+	return cfg, nil
+}
+
+// Option applies an autoexport configuration option.
+type Option interface {
+	apply(config) config
+}
+
+type optionFunc func(config) config
+
+func (fn optionFunc) apply(cfg config) config {
+	return fn(cfg)
+}
+
+// WithFallbackSpanExporter sets the fallback exporter to use when no exporter
+// is configured through the OTEL_TRACES_EXPORTER environment variable.
+func WithFallbackSpanExporter(exporter trace.SpanExporter) Option {
+	return optionFunc(func(cfg config) config {
+		cfg.fallbackExporter = exporter
+		return cfg
+	})
+}
+
+// NewTraceExporter returns a configured SpanExporter defined using the environment
+// variable OTEL_TRACES_EXPORTER, the configured fallback exporter via options or
+// a default OTLP exporter (in this order).
+func NewTraceExporter(ctx context.Context, opts ...Option) (trace.SpanExporter, error) {
```

Comment:
I am concerned about naming of  `NewTraceExporter` and `SpanExporter`.

1. How we will name the counterparts for metrics to handle `OTEL_METRICS_EXPORTER` in future? `NewMetricsExporter` and `MetricsExporter`?
2. By reading the names it is hard to understand which one uses the env var and which only uses the registry.
3. `TraceExporter` is never used in OTel Go API. I suggest keeping only SpanExporter to avoid confusion

Maybe we should rename `NewTraceExporter` to `SpanExporterFromEnv` (or `NewSpanExporter`)?

PS. I think more opinions are needed than just mine. Naming is hard 😉

Reply:
Side note: This is currently, the only reason why I have not approved the PR.

Reply:
Yes, I agree. I think `NewTraceExporter` was left over from the initial design. I've renamed to `NewSpanExporter`.

I also un-exported `registry.SpanExporter` func, users shouldn't try to load exporter's directly and should use the above func insteads.

@MrAlias Do you agree with this?

Reply:
> I also un-exported `registry.SpanExporter` func, users shouldn't try to load exporter's directly and should use the above func insteads.

We saw the case with the propagators that some users, like the collector, have a file configuration which includes a string defining the propagator they want to load. Having `registry.SpanExporter` would help those users not have to re-parse the string into an exporter here.

That said, we can add it afterwards like we did for the autoprop package.